### PR TITLE
Sleep wake cli

### PR DESCRIPTION
**Pull request by zerbitx**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #ENG-4157
resolves #ENG-4158
resolves #ENG-4159

**Please provide a short message that should be published in the vcluster release notes**
Sleep and wake features used across drivers (platform/helm) are more intelligent when the `--driver` flag is omitted, still providing useful information for failures if the wrong driver is explicitly used from the command.

### CODE REVIEW

File: cmd/vclusterctl/cmd/platform/add/cluster.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func (cmd *ClusterCmd) Run(ctx context.Context, args []string) error {
 
 	if os.Getenv("DEVELOPMENT") == "true" {
 		helmArgs = []string{
-			"upgrade", "--install", "loft", "./chart",
+			"upgrade", "--install", "loft", cmp.Or(os.Getenv("DEVELOPMENT_CHART_DIR"), "./chart"),
```
Comment (zerbitx) [inline cmd/vclusterctl/cmd/platform/add/cluster.go:146]: This came in handy when testing with `go run cmd/vclusterctl/main.go` and wanting to point to a different chart directory.  Its not related to the aim of the PR.

File: pkg/cli/add_vcluster_helm.go [hunk 2]
```diff
@@ -44,6 +48,46 @@ func AddVClusterHelm(
...
+			Question:     fmt.Sprintf("vCluster %s is asleep and needs to be reawakened before it can be added.  Would you like to wake it up and add it now?", vClusterName),
+			DefaultValue: "no",
+			Options: []string{
+				"no",
+				"yes",
+			},
+		})
+
+		if err != nil {
+			return fmt.Errorf("failed to capture your repsponse %w", err)
```
Comment (lizardruss) [inline pkg/cli/add_vcluster_helm.go:64]: nit: typo

File: pkg/cli/add_vcluster_helm.go [hunk 3]
```diff
@@ -44,6 +48,46 @@ func AddVClusterHelm(
...
+		}
+
+		// Bail if they don't want to wake the vCluster yet.
+		if answer == "no" {
+			return fmt.Errorf("please wakeup vCluster %s before adding it to the platform", vClusterName)
```
Comment (lizardruss) [inline pkg/cli/add_vcluster_helm.go:69]: This seems a little harsh... like they chose the wrong answer. Should we log a message like "The vCluster will be added the next time it is active."?

File: pkg/cli/add_vcluster_helm.go [hunk 4]
```diff
@@ -44,6 +48,46 @@ func AddVClusterHelm(
 		return err
 	}
 
+	// If the vCluster was paused with the helm driver, adding it to the platform will only create the secret for registration
+	// which leads to confusing behavior for the user since they won't see the cluster in the platform UI until it is resumed.
+	if vCluster.Annotations != nil && vCluster.Annotations[constants.PausedAnnotation] == "true" {
```
Comment (lizardruss) [inline pkg/cli/add_vcluster_helm.go:53]: Could you use lifecycle.IsPaused() here? or is there an import cycle / type conflict?

**Review by lizardruss — CHANGES_REQUESTED** [review]
Some nits, and a question about erroring when the user choosed not to wake up the vcluster during add.

File: pkg/cli/add_vcluster_helm.go (hunk 3 above)
Comment (zerbitx) [reply pkg/cli/add_vcluster_helm.go]: I said please... :)

File: pkg/platform/helper.go [hunk 5]
```diff
@@ -1112,7 +1112,7 @@ func WaitForVirtualClusterInstance(ctx context.Context, managementClient kube.In
 
 	if virtualClusterInstance.Status.Phase == storagev1.InstanceSleeping {
 		log.Info("Wait until vcluster wakes up")
-		defer log.Donef("Successfully woken up vcluster %s", name)
+		defer log.Donef("vCluster %s wakeup successful", name)
```
Comment (facchettos) [inline pkg/platform/helper.go:1115]: didn't we say that it should either be `vCluster instance` or `virtual cluster`?

File: pkg/lifecycle/lifecycle.go [hunk 6]
```diff
@@ -362,3 +362,7 @@ func scaleUpStatefulSet(ctx context.Context, kubeClient kubernetes.Interface, la
 
 	return true, nil
 }
+
+func IsPaused(annotations map[string]string) bool {
+	return annotations != nil && annotations[constants.PausedAnnotation] == "true"
```
Comment (facchettos) [inline pkg/lifecycle/lifecycle.go:367]: the check for nil isn't required, nil map will return 0 values for all keys

File: pkg/platform/sleepmode/sleepmode.go
Comment (facchettos) [inline pkg/platform/sleepmode/sleepmode.go:1]: I think i'd prefer to use methods on the objects directly, that'd be easier to read and discover

File: pkg/cli/find/find.go [hunk 7]
```diff
@@ -561,3 +561,17 @@ func GetPodStatus(pod *corev1.Pod) string {
 	}
 	return reason
 }
+
+func isPaused(v client.Object) bool {
+	annotations := v.GetAnnotations()
+	if annotations != nil && annotations[constants.PausedAnnotation] == "true" {
```
Comment (facchettos) [inline pkg/cli/find/find.go:567]: same for the nil check here and below

File: pkg/cli/pause_helm.go [hunk 8]
```diff
@@ -30,6 +30,11 @@ func PauseHelm(ctx context.Context, globalFlags *flags.GlobalFlags, vClusterName
 		return err
 	}
 
+	if vCluster.IsSleeping() {
+		log.Infof("vcluster %s/%s is already paused using the platform driver", globalFlags.Namespace, vClusterName)
```
Comment (lizardruss) [inline pkg/cli/pause_helm.go:34]: Maybe reword this to `vcluster %s/%s is already paused by the platform`. Calling attention to the driver may be confusing and implies that they put it to sleep when the platform could have due to sleep mode configuration.

**Review by lizardruss — CHANGES_REQUESTED** [review]
One suggestion for re-wording. If the wording has already been reviewed / approved then we can disregard.

**Review by lizardruss — APPROVED** [review]
lgtm
